File: Stack/주식가격.js
function solution(prices) {
    const stack = [[prices[0],0]];
    const answer= new Array(prices.length).fill(0)
    for(let i=1;i<prices.length;i++){
        while(stack.length && stack[stack.length-1][0]>prices[i]){
            const [price, index] = stack.pop()
            answer[index] = i-index
        }
        stack.push([prices[i],i])
    }
    const n = prices.length -1
    while(stack.length ){
        const [price, index] = stack.pop()
        answer[index] = n-index
    }

    return answer;
}

File: Stack/주식가격.py
def solution(prices):
    n = len(prices)
    answer = [0]*n
    stack=[[prices[0],0]]
    
    for i in range(1,n):
        while(len(stack) and stack[-1][0] >prices[i]):
            price,index=stack.pop()
            answer[index]=i-index
        stack.append([prices[i],i])
    
    for price,index in stack:
        answer[index]=n-1-index
    return answer
